osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -637,10 +637,16 @@ private List<TFileModel> createFileInfos(ArchiveReader reader, FileStore files)
         /// </summary>
         protected virtual string ImportFromStablePath => null;
 
+        /// <summary>
+        /// Checks for the existence of an osu-stable directory.
+        /// </summary>
+        protected virtual bool CheckStableDirectoryExists(StableStorage stableStorage) => stableStorage.ExistsDirectory(ImportFromStablePath);
+
         /// <summary>
         /// Select paths to import from stable. Default implementation iterates all directories in <see cref="ImportFromStablePath"/>.
         /// </summary>
-        protected virtual IEnumerable<string> GetStableImportPaths(Storage stableStoage) => stableStoage.GetDirectories(ImportFromStablePath);
+        protected virtual IEnumerable<string> GetStableImportPaths(StableStorage stableStoage) => stableStoage.GetDirectories(ImportFromStablePath)
```

Comment:
typo: `%s/stableStoage/stableStorage`. occurs in other places in this diff as well.